stack/rest/src/test/java/org/apache/usergrid/rest/test/resource2point0/endpoints/NamedResource.java

```diff
@@ -17,32 +17,35 @@
 package org.apache.usergrid.rest.test.resource2point0.endpoints;
 
 
+import org.apache.usergrid.rest.test.resource2point0.model.ApiResponse;
+import org.apache.usergrid.rest.test.resource2point0.model.Entity;
 import org.apache.usergrid.rest.test.resource2point0.model.QueryParameters;
 import org.apache.usergrid.rest.test.resource2point0.model.Token;
 import org.apache.usergrid.rest.test.resource2point0.state.ClientContext;
 
 import com.sun.jersey.api.client.WebResource;
 
-import java.util.HashMap;
+import javax.ws.rs.core.MediaType;
 import java.util.Iterator;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import java.util.UUID;
 
 
 /**
  * Base class that is extended by named endpoints.
  * The NamedResource stores the parent of the class, the context in which the class operates and then Name of this resource
  */
-public abstract class NamedResource implements UrlResource {
+public abstract class NamedResource<T extends NamedResource<T>> implements UrlResource {
 
-    protected final String name;
-    protected final ClientContext context;
+    protected String name;
+    protected ClientContext context;
     /* Stores the path of the parent that called it.
     i.e If we had a ApplicationResource ( an instance of a namedResource ) this would contain the OrganizationResource.
      */
-    protected final UrlResource parent;
+    protected UrlResource parent;
 
+    protected abstract T getThis();
```

Comment:
I guess the comment on clarity should probably go here explaining the purpose of the method.